source/Pgghost.h

```diff
@@ -0,0 +1,135 @@
+#ifndef PGGHOST_H
+#define PGGHOST_H
+
+// #include "../../Empirical/include/emp/math/Random.hpp"
+// #include "../../Empirical/include/emp/tools/string_utils.hpp"
+// #include <set>
+// #include <iomanip> // setprecision
+// #include <sstream> // stringstream
+#include "Host.h"
+#include "SymWorld.h"
+
+
+class PggHost: public Host {
+protected:
+  double sourcepool=0;
+
+
+public:
+  PggHost(emp::Ptr<emp::Random> _random, emp::Ptr<SymWorld> _world, emp::Ptr<SymConfigBase> _config,
+  double _intval =0.0, emp::vector<emp::Ptr<Organism>> _syms = {},
+  emp::vector<emp::Ptr<Organism>> _repro_syms = {},
+  std::set<int> _set = std::set<int>(),
+  double _points = 0.0) : Host(_random, _world, _config, _intval,_syms, _repro_syms, _set, _points) {}
+
+
+
+  PggHost(const PggHost &) = default;
+  PggHost(PggHost &&) = default;
+  PggHost() = default;
+ 
+  double GetPool() {return sourcepool;}
+  void SetPool(double _in) {sourcepool= _in;}
+  void AddPool(double _in) {sourcepool += _in;}
+
+
+  void DistribResources(double resources) {
```

Comment:
I think I had a comment about this before, but it must have gotten lost. You can replace most of this function with a call to Host::DistribResources(resources); Then you just need to add a for loop after it to handle the pool stuff (which is the only difference between this function and the Host's version)